packages/next/src/server/app-render/app-render.tsx

```diff
@@ -739,10 +739,7 @@ async function renderToHTMLOrFlightImpl(
       function onHeaders(headers: Headers): void {
         // Copy headers created by React into the response object.
         headers.forEach((value: string, key: string) => {
-          res.appendHeader(key, value)
-          if (!extraRenderResultMeta.extraHeaders) {
-            extraRenderResultMeta.extraHeaders = {}
-          }
+          extraRenderResultMeta.extraHeaders ??= {}
```

Comment:
would be good to update the comment a few lines above this to reflect the new behavior -- seems we're no longer copying headers into the response object